recipes/libmetalink/all/patches/0001-no-redefine-malloc.patch

```diff
@@ -0,0 +1,10 @@
+--- a/configure
```

Comment:
I think you an avoid a patch by adding `ac_cv_func_malloc_0_nonnull` to the configure args.
Or instead, try adding `STDC_HEADERS` to `self._autotools.defines`.

Check you `config.log`.
I think the following source fails to compile that prompted you to apply this patch:
```c
/* end confdefs.h.  */
#if defined STDC_HEADERS || defined HAVE_STDLIB_H
# include <stdlib.h>
#else
char *malloc ();
#endif

int
main ()
{
return ! malloc (0);
  ;
  return 0;
}
```
(This is due to [`AC_FUNC_MALLOC`](https://www.gnu.org/software/autoconf/manual/autoconf-2.60/html_node/Particular-Functions.html))

Reply:
yes, it's due to `AC_FUNC_MALLOC`, it breaks cross-compilation. I can't modify configure.ac and call autoreconf, because some m4 macros are missing in tarball.

Reply:
Ah yes, it contains `ac_fn_c_try_run`.
So you need to add `ac_cv_func_malloc_0_nonnull=yes` to the arg list.

Reply:
What m4 macros are missing?
We recently received a pr for `autoconf-archive`.
https://github.com/conan-io/conan-center-index/pull/6702

Reply:
Indeed, adding `ac_cv_func_malloc_0_nonnull=yes` works.

Reply:
> What m4 macros are missing?

`AM_PATH_XML2`

Reply:
https://gitlab.gnome.org/GNOME/libxml2/-/blob/master/libxml.m4

Looks like we need to install it too and add its directory to the `AUTOMAKE_CONAN_INCLUDES` environment variable.

Reply:
Yes, but it would be overkill for libmetalink, because:
- libxml2 is an optional backend
- libxml2 would have to be added to build requirements even though expat is the backend.

Reply:
You're right here.
Nonetheless, I've opened an issue at https://github.com/conan-io/conan-center-index/issues/6769